src/ui/pages/work/partusage/PartUsageDialog.js

```diff
@@ -202,124 +239,231 @@ function PartUsageDialog(props) {
-                    showError('Asset is not in any store.');
+                    showError(`Asset "${assetIDCode}" is not available in any store.`);
                     return undefined;
                 }
 
                 // Asset is in a store other than the selected (can happen by selecting asset from input history)
-                if (responseStoreCode !== formData.storeCode) {
-                    showError(`Asset is in a different store (${responseStoreCode}) than the selected.`);
+                if (responseStoreCode !== storeCode) {
+                    showError(`Asset "${assetIDCode}" is in a different store (${responseStoreCode}) than the selected.`);
                     return undefined;
                 }
 
                 return assetData;
 
-            // "RETURN" transaction type
-            } else if (formData.transactionType === transactionTypes[1].code) {
+            } else if (transactionType === returnType) {
                 if (responseStoreCode) {
-                    showError(`Asset is already in a store (${responseStoreCode}).`);
+                    showError(`Asset "${assetIDCode}" is already in a store (${responseStoreCode}).`);
                     return undefined;
                 }
 
                 return assetData;
             }
 
             // Something is wrong if we get here (user input or code-related)
-            showError('Asset does not follow business rules.');
+            showError(`Asset "${assetIDCode}" does not follow business rules.`);
             return undefined;
 
         } catch (error){
             handleError(error);
         }
     };
 
-    const handleAssetSelectedWithPart = (assetData) => {
-        const { bin, partCode } = assetData;
+    const handleAssetSelectedWithPart = async (assetData) => {
+        const { bin, partCode, lot } = assetData;
+        const { transactionType, storeCode } = formData;
+
+        // When in an issue transaction, bin loading will also load the lot list
+        // since assets have a single bin (which triggers lot loading).
+        await Promise.all([
+            loadBinList(bin, partCode),
+            transactionType === returnType
+                ? loadLotList(transactionType, lot, bin, partCode, storeCode)
+                : null,
+        ]);
+    };
 
-        updateFormDataProperty(FORM.BIN, '');
-        if (assetData) {
-            updateFormDataProperty(FORM.BIN, bin);
-            loadBinList(bin, partCode);
-        }
+    const handleAssetDifferentOrNoPart = async (assetData) => {
+        const { partCode } = assetData;
+
+        // Clear the part field so the user is aware that the selected asset is associated with a different part
+        resetFieldWithDesc(FORM.PART, FORM.PART_DESC);
+
+        const results = await Promise.all([
+            loadPartData(partCode),
+            handleAssetSelectedWithPart(assetData),
+        ]);
+
+        const partData = results[0];
```

Comment:
```
const [ partData ] = await Promise.all([
            loadPartData(partCode),
            handleAssetSelectedWithPart(assetData),
        ]);
```
I believe this is cleaner, what do you think?

Reply:
Also prefer that :+1: